### PR TITLE
Migrate notary-server repo

### PR DESCRIPTION
**Pull request by yuroitaki**
Addresses https://github.com/tlsnotary/notary-server/issues/15

### CODE REVIEW

File: notary-server/notary-server.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# !!! To use this file, please run docker run at the root level of this repository
```
Comment (yuroitaki) [inline notary-server/notary-server.Dockerfile:1]: Need to do this because we need to copy in dependency files from `tlsn` and `components` — mainly due to the restriction of docker build context

File: notary-server/notary-server.Dockerfile.dockerignore [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# exclude everything
```
Comment (yuroitaki) [inline notary-server/notary-server.Dockerfile.dockerignore:1]: Uses this prefix for this dockerignore file so that it will be used when the notary-server image is built; and changed to whitelisting the entire folder (e.g. `tlsn`, `components`) instead of individual files because that will be too much to specify as we are copying dependency files from core libs

**Review by heeckhau — CHANGES_REQUESTED** [review]
Looks good. But there are still some references to the notary-server repository.

I'll push a commit with some extra changes soon

File: .github/workflows/cd.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+on:
+  push:
+    tags:
+      - "[v]?[0-9]+.[0-9]+.[0-9]+*"
+
+env:
+  CONTAINER_REGISTRY: ghcr.io
+
+jobs:
+  build_and_publish_image:
```
[the remark this answers is not in the document]
Comment (yuroitaki) [reply .github/workflows/cd.yml:12]: I was actually thinking to make this generic so that this same job can be used for different docker build and push — since the first few steps of waiting for CI to pass and logging into GCR do not need to be repeated for different docker images in this same repo

File: .github/workflows/cd.yml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    name: Build and publish image
```
[the remark this answers is not in the document]
Comment (yuroitaki) [reply .github/workflows/cd.yml:13]: Same reason as above

File: .github/workflows/ci.yml [hunk 5]
```diff
@@ -61,14 +72,11 @@ jobs:
 
       - name: "Test"
         if: ${{ matrix.release != true }}
-        run: cargo test --lib --bins --tests --examples --workspace
+        run: RUST_LOG=DEBUG cargo test --lib --bins --tests --examples --workspace -- --nocapture
```
[the remark this answers is not in the document]
Comment (yuroitaki) [reply .github/workflows/ci.yml:75]: Good point, took a look and I think the TLS crate will print a lot of debug messages — also this addition is more for debugging purpose when one cannot determine what went wrong after rerunning the test locally as compared to in Github runner — so it's more of an edge case hence I will remove this modification

File: .github/workflows/cd.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (yuroitaki) [reply .github/workflows/cd.yml:12]: ^ignore this, I realised the `waiting for CI to pass` step needs to be specific for notary docker image in this case, because the `wait on check action` can only check for a specific package (i.e. notary-server) in the [matrix](https://github.com/tlsnotary/tlsn/blob/1c7273143ed8e2b8d05a5bc53e630b98f64badb4/.github/workflows/rust.yml#L19-L30) we used in `ci.yml`, instead of the overall test result of all the packages, more info [here](https://github.com/lewagon/wait-on-check-action#check-name) under the example where they use strategy matrix

File: .github/workflows/cd.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (yuroitaki) [reply .github/workflows/cd.yml:13]: Same update as above

**Review by themighty1 — APPROVED** [review]
nice work, thanks

**Review by heeckhau — APPROVED** [review]
